File: Leetcode/1400. Construct K Palindrome Strings.cpp
class Solution {
public:
    bool canConstruct(string s, int k) {
        vector<int> freq(26,0);
        if(s.size()<k){return false;}
        for(char c:s) freq[c-'a']++;
        for(int i=0;i<26;i++){
            if(freq[i]&1==1){
                if(k>0){
                    k--;
                }else{
                    return false;
                }
            }
        }
        return true;
    }
};

File: Leetcode/1400. Construct K Palindrome Strings.py
class Solution:
    def canConstruct(self, s: str, k: int) -> bool:
        freq=[0]*26
        if len(s)<k:
            return False
        for c in s:
            freq[ord(c)-ord('a')]+=1
        for i in range(26):
            if freq[i]&1 == 1:
                if k>0:
                    k-=1
                else:
                    return False
        return True
        
